lib/sproxyd.js

```diff
@@ -226,6 +233,11 @@ class SproxydClient {
         });
 
         let receivedResponse = false;
+        // keep track of the current endpoint prior to request.
```

Comment:
It's not clear to me why we need this change, could you try to clarify?

Reply:
the is to handle the issue with `batchDelete` failovers in which the bootstrap list is shifted multiple times, once per  parallel request, resulting in unnecessary shifts. keeping track of the node will help with determining if the bootstrap list should be shifted to prevent erroneous shifts